File: editor/en/[14]Longest Common Prefix.java
14
        Longest Common Prefix
        2023-01-02 10:39:06

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String str = strs[0];
        for (int i = 1; i < strs.length; i++) {
            while (strs[i].indexOf(str) != 0) {
                str = str.substring(0, str.length() - 1);
            }
        }
        return str;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[14]Longest Common Prefix.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""
        for x in zip(*strs):
            if len(set(x)) == 1:
                prefix += x[0]
            else:
                break

        return prefix

# leetcode submit region end(Prohibit modification and deletion)
